packages/cli/src/encodedFile.ts

```diff
@@ -0,0 +1,77 @@
+import buffer, { TranscodeEncoding } from 'buffer';
+import fs from 'fs';
+
+export default class EncodedFile {
+  private buf: Buffer;
+  private encoding?: TranscodeEncoding;
+  private bom0?: number;
+  private bom1?: number;
+
+  constructor(readonly path: fs.PathLike) {
+    const {buf, bom0, bom1, encoding} = EncodedFile.read(path);
+    this.buf = buf;
+    this.bom0 = bom0;
+    this.bom1 = bom1;
+    this.encoding = encoding;
+  }
+
+  toString(): string {
+    let ret;
+    if (this.encoding) {
+      // slice(1) to get rid of the BOM.
+      ret = this.buf.toString(this.encoding).slice(1);
+    } else {
+      ret = this.buf.toString();
+    }
+
+    return ret;
+  }
+
+  // Encode the given string with our encoding and write it to path
+  // @param {string} str the string to write
+  // @param {PathLike} [path] the path to write the string to, defaults to this.path
+  write(str: string, path?: fs.PathLike) {
+    let buf = Buffer.from(str);
+
+    if (this.encoding) {
+      buf = buffer.transcode(buf, 'utf8', this.encoding);
```

Comment:
Wouldn't it make more sense to provide the encoding to `Buffer.from` instead of immediately transcoding it?

Reply:
> My reading of the doc for Buffer.from is that the encoding option describes what's in the source string, not the desired encoding. So, the transcoding is necessary to get to the target encoding.

I've determined that I was mistaken, and you were right: the `encoding` passed to `Buffer.from` describes the encoding for the bytes that wind up in `buf`. I'll get rid of the call to `transcode`.

UTF-16 characters in the test files come through correctly, so I don't think there are any other changes that need to be made.